public/app/plugins/datasource/loki/result_transformer.ts

```diff
@@ -1,155 +1,16 @@
-} from '@grafana/ui';
-
-export function processEntry(
-  entry: LogsStreamEntry,
-  labels: string,
-  parsedLabels: Labels,
-  uniqueLabels: Labels,
-  search: string
-): LogRowModel {
-  const { line } = entry;
-  const ts = entry.ts || entry.timestamp;
-  // Assumes unique-ness, needs nanosec precision for timestamp
-  const key = `EK${ts}${labels}`;
-  const time = moment(ts);
-  const timeEpochMs = time.valueOf();
-  const timeFromNow = time.fromNow();
-  const timeLocal = time.format('YYYY-MM-DD HH:mm:ss');
-  const logLevel = getLogLevel(line);
-  const hasAnsi = hasAnsiCodes(line);
-
-  return {
-    key,
-    logLevel,
-    timeFromNow,
-    timeEpochMs,
-    timeLocal,
-    uniqueLabels,
-    hasAnsi,
-    entry: hasAnsi ? ansicolor.strip(line) : line,
-    raw: line,
-    labels: parsedLabels,
-    searchWords: search ? [search] : [],
-    timestamp: ts,
-  };
-}
-
-export function mergeStreamsToLogs(streams: LogsStream[], limit = DEFAULT_MAX_LINES): LogsModel {
-  // Unique model identifier
-  const id = streams.map(stream => stream.labels).join();
-
-  // Find unique labels for each stream
-  streams = streams.map(stream => ({
-    ...stream,
-    parsedLabels: parseLabels(stream.labels),
-  }));
-  const commonLabels = findCommonLabels(streams.map(model => model.parsedLabels));
-  streams = streams.map(stream => ({
-    ...stream,
-    uniqueLabels: findUniqueLabels(stream.parsedLabels, commonLabels),
-  }));
-
-  // Merge stream entries into single list of log rows
-  const sortedRows: LogRowModel[] = _.chain(streams)
-    .reduce(
-      (acc: LogRowModel[], stream: LogsStream) => [
-        ...acc,
-        ...stream.entries.map(entry =>
-          processEntry(entry, stream.labels, stream.parsedLabels, stream.uniqueLabels, stream.search)
-        ),
-      ],
-      []
-    )
-    .sortBy('timestamp')
-    .reverse()
-    .value();
-
-  const hasUniqueLabels = sortedRows && sortedRows.some(row => Object.keys(row.uniqueLabels).length > 0);
-
-  // Meta data to display in status
-  const meta: LogsMetaItem[] = [];
-  if (_.size(commonLabels) > 0) {
-    meta.push({
-      label: 'Common labels',
-      value: commonLabels,
-      kind: LogsMetaKind.LabelsMap,
-    });
-  }
-  if (limit) {
-    meta.push({
-      label: 'Limit',
-      value: `${limit} (${sortedRows.length} returned)`,
-      kind: LogsMetaKind.String,
-    });
-  }
-
-  return {
-    id,
-    hasUniqueLabels,
-    meta,
-    rows: sortedRows,
-  };
-}
-
-export function logStreamToSeriesData(stream: LogsStream): SeriesData {
+export function logStreamToSeriesData(stream: LokiLogsStream): SeriesData {
   let labels: Labels = stream.parsedLabels;
   if (!labels && stream.labels) {
     labels = parseLabels(stream.labels);
   }
   return {
     labels,
-    fields: [{ name: 'ts', type: FieldType.time }, { name: 'line', type: FieldType.string }],
+    fields: [{ name: 'time', type: FieldType.time }, { name: 'message', type: FieldType.string }],
```

Comment:
Decided to use time and message to make it more datasource agnostic. Also time is often used for metrics so thought that can be a good thing.

Reply:
👍

Reply:
time is good -- but it should not matter what name you use since you the type is 'time' and on the other end you should use `getFirstTimeField`

Reply:
Have fixed this to be name-independent and use time and string field as requirement instead.